### PR TITLE
Port activation range improvement patch

### PR DESCRIPTION
**Pull request by kickash32**
Will require review and testing to verify the patch works as it is intended due to big codebase changes from 1.13 and 1.15. Patch currently compiles and seem to run without breaking the game.

### CODE REVIEW

File: Spigot-Server-Patches/0416-Activation-Range-Improvements.patch [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+ import net.minecraft.server.EntityLiving;
++import net.minecraft.server.EntityLlama;
+ import net.minecraft.server.EntityMonster;
+ import net.minecraft.server.EntityProjectile;
+ import net.minecraft.server.EntityRaider;
+@@ -28,6 +30,7 @@ import net.minecraft.server.EntitySlime;
+ import net.minecraft.server.EntityTNTPrimed;
+ import net.minecraft.server.EntityThrownTrident;
+ import net.minecraft.server.EntityVillager;
++import net.minecraft.server.EntityWaterAnimal;
```
Comment (kickash32) [inline Spigot-Server-Patches/0416-Activation-Range-Improvements.patch]: missing 3 paper comments

File: Spigot-Server-Patches/0416-Activation-Range-Improvements.patch [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+ import net.minecraft.server.EntityWither;
+ import net.minecraft.server.MathHelper;
+ import net.minecraft.server.MinecraftServer;
+@@ -42,7 +45,8 @@ public class ActivationRange
+         MONSTER,
+         ANIMAL,
+         RAIDER,
+-        MISC;
++        MISC,
++        WATER;
```
Comment (kickash32) [inline Spigot-Server-Patches/0416-Activation-Range-Improvements.patch]: reduce diff

**Comment by kickash32** [comment]
Patch is done, but it needs review as I'm not 100% sure I ported it correctly. Lot of unmapped code was changed so I did my best at matching it.

**Review by electronicboy — APPROVED** [review]

**Comment by kickash32** [comment]
Should be good now @electronicboy.

**Pull request merged**